enterprise/lms_api.py

```diff
@@ -168,10 +168,22 @@ def get_course_enrollment(self, username, course_id):
             '{username},{course_id}'.format(username=username, course_id=course_id)
         )
         try:
-            return endpoint.get()
+            result = endpoint.get()
         except HttpNotFoundError:
-            LOGGER.error('course enrollment details not found for username=%s, course=%s', username, course_id)
+            # This enrollment data endpoint returns a 404 if either the username or course_id specified isn't valid
+            LOGGER.error(
+                'course enrollment details not found for invalid username or course; username=%s, course=%s',
+                username,
+                course_id
+            )
             return None
+        # This enrollment data endpoint returns an empty string if the username and course_id is valid, but there's
+        # no matching enrollment found
+        if len(result) == 0:
```

Comment:
@douglashall The result can either be a byte string or a dictionary. I'd be cool with `if not result` if you are. I just opted to check for emptiness in a slightly more specific way that represents a subset of what `not result` will evaluate to true with. What do you prefer?